lib/index.js

```diff
@@ -13,14 +13,22 @@ function compareColors(nodeColor, targetColor) {
     .every((c, i) => parseFloat(c.value) === targetColor[i]);
 }
 
-function applyColorFunction(node, initialColor, finalColor) {
+function applyColorFunction(node, initialColor, finalColor, alphaChannel) {
   const {name} = node;
   const nodeColor = node.children
     .filter(item => item.type === 'Number' || item.type === 'Percentage')
     .toArray();
 
-  if (compareColors(nodeColor, initialColor[name])) {
+  const cloneNodeColor = [...nodeColor];
+  const sliceNodeColor = cloneNodeColor.slice(0, 3);
+  const sliceInitialColor = initialColor[name].slice(0, 3);
```

Comment:
You don’t need to slice `targetColor` here because `compareColors` function checks only the color components from `nodeColor` (see line 13).

On the other hand, I would provide an option to `compareColors` and name it `preserveAlphaChannel`.

Reply:
Okey